src/server/editorServices.ts

```diff
@@ -642,7 +646,6 @@ namespace ts.server {
                 return undefined;
             }
             if (isInferredProjectName(projectName)) {
-                this.ensureProjectStructuresUptoDate();

```

Comment:
so why is this removed?

Reply:
From the commit info:
There is no need to ensure project structure when trying to get inferred project
The project operations will validate if its valid anyways and throw exception accordingly. So this is just saving us check which is quite costly.

Reply:
I thought we wanted to make sure that the file is not claimed by a configured project that was out-of-date before considering inferred projects.. is not that why we added this in the first place?

Reply:
This isnt used for that though. That part is still valid. This findProject is just trying to lookup project with given name, and is used by session to get the project with given name passed in or by updateTypings to get the project for which the typings are updated. Both of these do not need to update the whole project structure. They already have project name, so just passing on the project makes sense.  (eg. when trying to set typings for inferred project, we keep on doing this too frequently if the .js file is in inferred project)